lesson_7/mosut_sorin/hw_lesson7_decorators_mosut_sorin3.py

```diff
@@ -0,0 +1,70 @@
+'''
+
+    Given a set of print methods:
+
+print_registry = []
+
+
+def greet(name):
+    return "Greetings {}!".format(name)
+
+
+def say_hello(name):
+    return "Hello {}!".format(name)
+
+
+def say_goodbye(name):
+    return "Goodbye {}!".format(name)
+
+Create a decorator called register that will update a list called print_registry with all the decorated functions names.
+
+print_registry = []
+
+@register
+def greet(name):
+    return "Greetings {}!".format(name)
+
+
+def say_hello(name):
+    return "Hello {}!".format(name)
+
+@register
+def say_goodbye(name):
+    return "Goodbye {}!".format(name)
+
+print(print_registry)
+['greet', 'say_goodbye']
+
+'''
+
+print_registry = []
+
+def register(fnc):
+    def iner_register(args):
+        fnc_name = str(fnc)
+        fnc_name = fnc_name.split(' ')
+        print_registry.append(fnc_name[1])
+
+    return iner_register
```

Comment:
Is the inner function really needed?
Justify the answer, whichever that might be

Reply:
If we want only to get this list ['greet', 'say_goodbye'] , is no need of a inner function becouse the list will be created at the steps @register , bun without the inner functions we can not get any result from the decorated functions.

Reply:
Are you sure about that?

Reply:
You say to use only the registry() function , with no other nested function inside it?

Reply:
Yap

Reply:
You are wright , it works like this:

print_registry = []

def register(fnc):
        fnc_name = str(fnc).split(' ')
      #  fnc_name = fnc_name.split(' ')
        print_registry.append(fnc_name[1])
        return fnc
    
@register
def greet(name):
    return "Greetings {}!".format(name)

def say_hello(name):
    return "Hello {}!".format(name)

@register
def say_goodbye(name):
    return "Goodbye {}!".format(name)

print(greet('Sorin'))
print(say_hello('Sorin'))
print(say_goodbye('Sorin'))
print(print_registry)

Reply:
You are wright , it works like this:

print_registry = []

def register(fnc):
        fnc_name = str(fnc).split(' ')
        print_registry.append(fnc_name[1])
        return fnc
    
@register
def greet(name):
    return "Greetings {}!".format(name)

def say_hello(name):
    return "Hello {}!".format(name)

@register
def say_goodbye(name):
    return "Goodbye {}!".format(name)

print(greet('Sorin'))
print(say_hello('Sorin'))
print(say_goodbye('Sorin'))
print(print_registry)